File: web/src/app/author.ts
import {Sequence} from "./sequence";
import {Book} from "./book";
import {AuthorBook} from "./author-book";

export class Author {

  id: number;
  name: string;
  descr: string;

  sequences: Array<Sequence>;
  booksNoSequence: Array<AuthorBook>;

}

File: common/src/main/java/com/patex/forever/model/Author.java
package com.patex.forever.model;


import lombok.Data;

import java.time.Instant;
import java.util.Collection;
import java.util.List;
import java.util.stream.Collectors;
import java.util.stream.Stream;

import static java.util.Collections.emptyList;

@Data
public class Author {

    private Long id;

    private String name;

    private List<Book> booksNoSequence = emptyList();

    private List<Sequence> sequences = emptyList();

    private String descr;

    private Instant updated;

    public List<Book> getBooks() {
        return Stream.concat(sequences.stream().map(Sequence::getBooks).
                        flatMap(Collection::stream).map(SequenceBook::getBook),
                booksNoSequence.stream()).collect(Collectors.toList());
    }
}